modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -492,6 +494,23 @@ const FrontendDataSet = ({
 						style={style}
 						{...currentViewProps}
 					/>
+				) : emptyState ? (
+					<ClayEmptyState
+						description={Liferay.Language.get(
+							'start-creating-one-to-show-your-data'
+						)}
+						imgSrc={`${themeDisplay.getPathThemeImages()}/states/empty_state.gif`}
+						title={emptyState.title}
+					>
+						<ClayButton
+							displayType="secondary"
+							onClick={() => {
+								creationMenu.primaryItems[0].onClick(loadData);
+							}}
```

Comment:
Hey Marko, is this the type of change you asked for? https://github.com/liferay-frontend/liferay-portal/pull/3399/commits/2203d0d10ecb51a8602fa4baf51e6679318adb90

Reply:
Not quite. I like this version more then the previous one, but the questions remain:
- What if there are multiple items or no items in menu? In the current code, we have a button prop. We are completely decoupling creation menu and button definition. Do we want that? I am not sure that we do. My understanding of the task is to duplicate functionality of the creation menu.
- What if users want to set a custom message, but not have a button? The current code breaks with NPE.

Reply:
> What if there are multiple items or no items in menu?

This was not in the design so it could be a little out of scope but I've added a new commit to support this case https://github.com/liferay-frontend/liferay-portal/pull/3399/commits/a9b78c7ce9b72e2060d74f6d645dbd9f054aaaef#diff-8bc70260aeeec91969d2ac7da9174557ebb931b6f352d71b4e4c066e2e6449abR514 . I was trying to replace this part with a <CreationMenu> instance but at the moment it looks like:

<img width="627" alt="image" src="https://github.com/liferay-frontend/liferay-portal/assets/46778114/eae3a1f8-d962-4853-8e4b-ea02e108715a">

![fdse-creationmenu](https://github.com/liferay-frontend/liferay-portal/assets/46778114/4d275a13-cf35-4b7e-947b-0bc20021c47d)

So it needs more work (changing the CreationMenu component). - Do we want to proceed on this?

>We are completely decoupling creation menu and button definition. Do we want that? I am not sure that we do. My understanding of the task is to duplicate functionality of the creation menu. Note: from the perspective of API, it is completely irrelevant that you as setting the same object to two props.

In this case I think it makes sense to keep them separated, since we might want to add the creationMenu into the `<ManagementBar>` but we might not want the same options or the button itself in the `<ClayEmptyState>`

Reply:
Since the `<CreationMenu>` looked the most reasonable option, I went ahead and integrated it https://github.com/liferay-frontend/liferay-portal/pull/3399/commits/118736128a2a4eb1e89f727ec44bef6b675928d4